### PR TITLE
More complicated test case for BST depth() method test

### PR DESCRIPTION
**Pull request by RepNazar**
I created BST depth() method implementation, that passed through tests, but wasn't working correct (you can look at it in try1 branch of my forked repo). 
So i added more complicated test case, which my implementation fails, and decided to contribute it into main branch.

### CODE REVIEW

File: 2-0-data-structures-and-algorithms/2-2-6-binary-search-tree/src/test/java/com/bobocode/cs/BinarySearchTreeTest.java [hunk 1]
```diff
@@ -260,7 +260,21 @@ void inorderTraversal() {
...
+                 * .../......\
+                 * ..4........20
+                 * ...\........\
+                 * ....5........24
+                 * ..............\
+                 * ...............26
+                 * ................\
+                 * .................30
+                 */
+                arguments(new Integer[]{10,6,18,4,8,12,20,5,24,26,30},5));
```
Comment (tboychuk) [inline 2-0-data-structures-and-algorithms/2-2-6-binary-search-tree/src/test/java/com/bobocode/cs/BinarySearchTreeTest.java:277]: @RepNazar could you please reformat this line to **add missing spaces**?

**Comment by tboychuk** [comment]
@RepNazar thanks a lot! 👏

File: 2-0-data-structures-and-algorithms/2-2-6-binary-search-tree/src/test/java/com/bobocode/cs/BinarySearchTreeTest.java (hunk 1 above)
Comment (RepNazar) [reply 2-0-data-structures-and-algorithms/2-2-6-binary-search-tree/src/test/java/com/bobocode/cs/BinarySearchTreeTest.java]: Sorry, forgot to press IDEA shortcut😅

**Comment by tboychuk** [comment]
@RepNazar could you also please **squash your commits**?

File: 2-0-data-structures-and-algorithms/2-2-6-binary-search-tree/src/test/java/com/bobocode/cs/BinarySearchTreeTest.java [hunk 2]
```diff
@@ -260,7 +260,21 @@ void inorderTraversal() {
...
                  * ..........5
                  */
-                arguments(new Integer[]{1, 2, 3, 4, 5}, 4));
+                arguments(new Integer[]{1, 2, 3, 4, 5}, 4),
+                /*
+                 * ......10
+                 * ...../..\
+                 * ....6....18
+                 * .../......\
+                 * ..4........20
```
Comment (tboychuk) [inline 2-0-data-structures-and-algorithms/2-2-6-binary-search-tree/src/test/java/com/bobocode/cs/BinarySearchTreeTest.java:269]: @RepNazar I just realized that **8, and 12 are missing**

**Review by tboychuk — CHANGES_REQUESTED** [review]

File: 2-0-data-structures-and-algorithms/2-2-6-binary-search-tree/src/test/java/com/bobocode/cs/BinarySearchTreeTest.java (hunk 2 above)
Comment (RepNazar) [reply 2-0-data-structures-and-algorithms/2-2-6-binary-search-tree/src/test/java/com/bobocode/cs/BinarySearchTreeTest.java]: Also noticed, that some "Always go right" implementation will pass all tests including mine. So I changed it

**Review by tboychuk — APPROVED** [review]
Thanks a lot!

**Comment by RepNazar** [comment]
My pleasure. Should i also create PR into completed branch?

**Pull request merged**